ambry-clustermap/src/main/java/com.github.ambry.clustermap/HardwareLayout.java

```diff
@@ -196,6 +200,16 @@ public Datacenter findDatacenter(String datacenterName) {
     return null;
   }
 
+  /**
+   * Finds Datacenter by id
+   *
+   * @param id id of datacenter to find
+   * @return Datacenter or null if not found.
+   */
+  public Datacenter findDatacenter(byte id) {
+    return datacenterById.get(id);
```

Comment:
sorry didn't have a chance to look at this earlier but there is a bug here .- id can be -1 for the older ids (before datacenter ID was introduced)

Reply:
This method signature says that it would return null if the datacenter is not found. This can happen not just for the -1 case, but also in the case where datacenters are decommissioned. As such I don't think there is a bug here. As for handling the null case, I see that #928 was introduced, but I don't think that change is necessary. See my comment there.